### PR TITLE
cmd/bootnode: fixes circuit reservation resource issues

### PR DESCRIPTION
**Pull request by dB2510**
Adds flags to modify max reservations per peer and relay timeout.

category: bug
ticket: #580

### CODE REVIEW

File: cmd/bootnode.go [hunk 1]
```diff
@@ -56,17 +58,19 @@ func newBootnodeCmd(runFunc func(context.Context, BootnodeConfig) error) *cobra.
...
-	flags.BoolVar(autoP2PKey, "auto-p2pkey", true, "Automatically create a p2pkey (ecdsa private key used for p2p authentication and ENR) if none found in data directory")
-	flags.BoolVar(p2pRelay, "p2p-relay", true, "Enable libp2p tcp host providing circuit relay to charon clusters")
+func bindBootnodeFlag(flags *pflag.FlagSet, config *BootnodeConfig) {
+	flags.StringVar(&config.HTTPAddr, "bootnode-http-address", "127.0.0.1:16005", "Listening address (ip and port) for the bootnode http server serving runtime ENR")
+	flags.BoolVar(&config.AutoP2PKey, "auto-p2pkey", true, "Automatically create a p2pkey (ecdsa private key used for p2p authentication and ENR) if none found in data directory")
+	flags.BoolVar(&config.P2PRelay, "p2p-relay", true, "Enable libp2p tcp host providing circuit relay to charon clusters")
+	flags.IntVar(&config.MaxResPerPeer, "max-reservations", 8, "Updates max reservations per peer")
+	flags.IntVar(&config.RelayTimeout, "relay-timeout", 2, "Updates default timout of a relay circuit reservation. (in minutes)")
```
Comment (corverroos) [inline cmd/bootnode.go:73]: can do `flags.Duration` which handles any duration unit

**Review by corverroos — APPROVED** [review]

File: cmd/bootnode.go [hunk 2]
```diff
@@ -56,17 +58,19 @@ func newBootnodeCmd(runFunc func(context.Context, BootnodeConfig) error) *cobra.
...
 
-func bindBootnodeFlag(flags *pflag.FlagSet, httpAddr *string, autoP2PKey, p2pRelay *bool) {
-	flags.StringVar(httpAddr, "bootnode-http-address", "127.0.0.1:16005", "Listening address (ip and port) for the bootnode http server serving runtime ENR")
-	flags.BoolVar(autoP2PKey, "auto-p2pkey", true, "Automatically create a p2pkey (ecdsa private key used for p2p authentication and ENR) if none found in data directory")
-	flags.BoolVar(p2pRelay, "p2p-relay", true, "Enable libp2p tcp host providing circuit relay to charon clusters")
+func bindBootnodeFlag(flags *pflag.FlagSet, config *BootnodeConfig) {
+	flags.StringVar(&config.HTTPAddr, "bootnode-http-address", "127.0.0.1:16005", "Listening address (ip and port) for the bootnode http server serving runtime ENR")
+	flags.BoolVar(&config.AutoP2PKey, "auto-p2pkey", true, "Automatically create a p2pkey (ecdsa private key used for p2p authentication and ENR) if none found in data directory")
+	flags.BoolVar(&config.P2PRelay, "p2p-relay", true, "Enable libp2p tcp host providing circuit relay to charon clusters")
+	flags.IntVar(&config.MaxResPerPeer, "max-reservations", 8, "Updates max reservations per peer")
```
Comment (xenowits) [inline cmd/bootnode.go:72]:
```suggestion
	flags.IntVar(&config.MaxResPerPeer, "max-reservations", 8, "Updates max circuit reservations per peer")
```

File: cmd/bootnode.go (hunk 1 above)
Comment (xenowits) [inline cmd/bootnode.go:73]:
```suggestion
	flags.IntVar(&config.RelayTimeout, "relay-timeout", 2, "Update default timeout (in minutes) of a relay circuit reservation")
```

**Review by xenowits — APPROVED** [review]
Looks good!

File: cmd/bootnode.go (hunk 1 above)
Comment (dB2510) [reply cmd/bootnode.go]: didn't used that so that i can be specific of assigning it in minutes

**Pull request merged**